sound/soc/intel/boards/sof_board_helpers.c

```diff
@@ -427,6 +427,7 @@ int sof_intel_board_set_dai_link(struct device *dev, struct snd_soc_card *card,
 	int ret;
 	int ssp_hdmi_in = 0;
 	unsigned long link_order, link;
+	unsigned long link_ids, be_id;
```

Comment:
what's the difference between 'link' and 'be_id'?

Reply:
'link' is the current dai link to add, and 'be_id' is for id field of this link

Reply:
not following the logic and when the 'link' value might be different to the be_id....

With the holiday break we've all reset our memories and revisiting this PR with no context at all...